uimaj-core/src/main/java/org/apache/uima/cas/impl/BinaryCasSerDes.java

```diff
@@ -973,6 +974,8 @@ private SerialFormat binaryDeserialization(Header h) {
         msg = e.toString();
       }
       throw new CASRuntimeException(CASRuntimeException.BLOB_DESERIALIZATION, msg);
+    } finally {
+        baseCas.forAllViews(view -> view.set_deserialized_doc_annot_not_indexed(null));

```

Comment:
A comment could be nice here explaining why this is done.

Reply:
Why does the setter not use the typical Java naming conventions?

Reply:
I'll add a comment.  re: naming conventions: I'm trying out "snake" case as an alternative to "camel" case.  I seen reports that for non-native English readers, it's a bit easier to follow.

Reply:
Snake-case is typical for Python - but in Java code, I'd rather opt to sticking with the traditional Java conventions.